Relative URLs via {{ site.baseurl }}

SeanKilleen opened the pull request:
This is no longer a work in progress and can be reviewed / discussed for merge.

This resolves #185.
This resolves #157.
## Goal

The goal of this PR is to move URLs to relative links where applicable, so that the links will function in relative environments such as another host or when built via Jekyll locally.
## Overall changes to allow for relative URLs:
- **Use the site.baseurl variable**: All markdown files were updated to use `{{ site.baseurl }}` as the first path of their path. This should be the original combination of host. 
  - Note: This can be set in the `_config.yml` file, though in our case we're good to leave it as the default (I believe -- see "Potential Issues" below)
## Potential Issues
- This does not affect `.html` files, which will need to have the full URL still for the time being. I believe there's a way around this, but it should be addressed as a separate PR.
- This PR makes the assumption that Github knows the baseurl includes `/joining-18f`, which I believe is correct. However, if I'm incorrect, this means that we'd have to add the `baseurl: "/joining-18f"` to the `_config.yml` file. It would be a quick change, but you may want to consider this before pushing as it could result in some minor downtime if I'm wrong.
## Work List

After a quick review, it looks like the following pages contain links that could be made relative:
- [x] index.md
- [x] how-to-apply.md
- [x] ~~readme.md~~ 
  - (leaving static URL here for cloning, etc. since it's meant to point to the official site).
- [x] acquisition-specialist.md
- [x] data-architect.md
- [x] design-and-product-strategist.md
- [x] director-of-biz-dev-and-products.md 
- [x] director-of-technical-architecture.md 
- [x] engagement-manager.md 
- [x] practical-dev.md
- [x] resource_manager.md
- [x] technical-architect.md

Commit:
Update roles-and-teams.md

Final step in #124
Updated to reflect the new priority roles.

Closing PR #167 as content changes are now captured here.

SeanKilleen committed:
Make URLs relative via site.baseurl

SeanKilleen commented:
Shoot. Just realized this site is a sub-site so standard relative URLs won't work (because it's relative from your sub-site, not the domain). 

I'll likely end up adding a base URL that's environment-specific in the _config.yml and parsing links from there.

SeanKilleen commented:
@fureigh I'm correct in my above realization, right? pages.18f.gov is a site which has several other sites off of it, so the relative URLs will really have to be relative to a base URL that's defined in the config. 

Just making sure I'm not missing anything there. (it's not hard to do -- I'll look up how to have it change accordingly based on the environment to make the local build process easier and avoid messing with the configuration.)

SeanKilleen commented:
[ **This comment no longer applies and was based upon some earlier assumptions that were incorrect.** ]

OK, as of now, the baseurl expects to exist at `[host:port]/joining-18f`. So when you run `bundle exec jekyll serve`, the site will actually be served at `http://localhost:4000/joining-18f`.

This can be overridden by running `bundle exec jekyll serve --baseurl ''` which will set the base URL to a blank. 

Pages going forward will be moved to use `{{ site.baseurl }}/relative/path`, which will allow all the links to be relative while confirming to the appropriate structure for whatever site they're on.

I'll also have to go clean up some of the other relative links I touched previously, since they were handled incorrectly.

SeanKilleen commented:
@fureigh this should now be ready for review. See my notes on potential issues.

Commit:
minor formatting edit

fureigh committed:
Various copy edits

Commit:
roles and teams change

Commit:
Added new roles and links to profiles

Commit:
Fix plurality with SRE

Commit:
Active to true

Commit:
active to true

Commit:
active to true

Commit:
active to true

fureigh committed:
Update text of role selection question

fureigh committed:
Change role selection question from text to select

1. Change role selection question to be a select instead of a textarea.
2. Make the role selection question required.
3. Add the specified dropdown options.

fureigh committed:
Add question: 'Tell us why you're interested'

Now that the role selection question is a 'select' field, the 'Tell us why you're interested' question is intended to encourage applicants to provide more context for their choice.

fureigh committed:
Remove rogue 'ss-q-title's

Remove orphaned <div> tags. Thanks for spotting this, @wslack!

fureigh committed:
Link to 'roles' page instead of 'who-we're-hiring'

Addresses @rooneywp's comment.

fureigh commented:
Thanks for this, @SeanKilleen! Reviewing today.

SeanKilleen commented:
@fureigh no problem! Let me know if you have any questions.

fureigh commented:
Thanks again for your work on this, @SeanKilleen!

To the potential issues you raised:
- :ok_hand: to leaving this for a separate PR, and
- You're correct about `site.baseurl`; no changes to `_config.yml` are needed. ([Further documentation in case you're curious](https://pages.18f.gov/guides-template/add-a-new-page/).)

Given the consistency of the edits and the back-and-forth about which files proved relevant, I do think a squash is appropriate here. Would you be willing to do an interactive rebase or would you prefer I squash merge this PR?

SeanKilleen commented:
@fureigh sorry, just seeing this now. Nice guide template! Appreciate the extra confirmation.

I should be able to do the interactive rebase -- I'll try to get on it right now but if I get interrupted or it takes too long, feel free to squash merge, I won't be offended. :)

SeanKilleen committed:
Additional URL changes for roles and teams page

I think changes that I merged back in from main meant that some
additional updates had to be made.

SeanKilleen commented:
@fureigh so.....I think I may have done that incorrectly. My process, so we can see where I went wrong:
- I brought in the updates from the main branch, which in retrospect I probably should not have done first. 
- I re-did some links in one of the MD files that had been updated 
- Then I ran a `git rebase -i HEAD~18`, squashing my commits and picking the rest of the commits including those ahead of mine that were made by others. I updated the commit message and pushed things out.

My expectation was that it would be alright to merge this this automatically because it contained all the changes in the master branch, with mine squashed, and then my additional corrections from after I merged the main branch back in.

However, I'm seeing a merge conflict here instead, so I'm thinking I messed something up.

Any thoughts on where I've gone wrong? I thought I understood interactive rebasing but this has me questioning things..

SeanKilleen commented:
@fureigh by the way, if it's easier, I can just re-do the changes in a separate PR now that I know what I'm doing on it. No need to have a lot of overhead on such a small change in order to merge this. It's simple to repeat. 

Let me know if that's the way you want to go on it.

fureigh commented:
@SeanKillen Simplest may be to git pull from master, make a fresh branch from that, squash merge these changes into your new branch and submit a new PR linked to this one.

(Pardon the delayed reply; been sick these past few days. GitHub should have an out-of-office setting!)

fureigh commented:
(@SeanKillen --> @SeanKilleen)

SeanKilleen commented:
Closing in favor of #202, which has all of the features and none of the cruft.
